src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -689,6 +689,17 @@ public override long Position
 
         public override Task FlushAsync(CancellationToken cancellationToken) => InnerStream.FlushAsync(cancellationToken);
 
+        public virtual Task NegotiateClientCertificateAsync(CancellationToken cancellationToken = default)
+        {
+            ThrowIfExceptionalOrNotAuthenticated();
+            if (RemoteCertificate != null)
+            {
+                throw new InvalidOperationException(SR.net_ssl_certificate_exist);
+            }
```

Comment:
Is this necessary? Could it return a completed task instead?

What if NegotiateClientCertificateAsync had already been called and no certificate was available? We would not want to renegotiate again, a completed task would also be appropriate then.

Reply:
We may. I wanted to make difference between case when client does not return certificate vs platform were this will never work. `NegotiateClientCertificateAsync`  can be called only once per discussion in #49346 e.g. if client refuses to give certificate we would not try it again. 

https://github.com/dotnet/runtime/issues/49346#issuecomment-823527996

>> At most one client certificate negotiation should happen for the lifetime of an SslStream instance, even if a prior call returned null.

>I assume this means that the API will immediately throw if either (a) we already have a client cert or (b) we previously requested one and did not get one. Correct?

Reply:
OK, sounds reasonable.